src/main/java/com/exasol/adapter/dynamodb/mapping/ToJsonPropertyToColumnMapping.java

```diff
@@ -48,4 +48,17 @@ public ColumnMapping withNewExasolName(final String newExasolName) {
         return new ToJsonPropertyToColumnMapping(newExasolName, this.getPathToSourceProperty(),
                 this.getLookupFailBehaviour());
     }
+
+    @Override
+    public boolean equals(final Object other) {
+        if (!(other instanceof ToJsonPropertyToColumnMapping)) {
+            return false;
+        }
+        return super.equals(other);
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode() + 623;
+    }
```

Comment:
I addded this random number because a ToJsonPropertyToColumnMapping is not equal to an ToStringPropertyToColumnMapping even so all their parameters are equal. Right now the ToStringPropertyToColumnMapping has additional fields what makes it diffent to the abstract class. In the future there could however be a ToNumberPropertyToColumnMapping that just like the ToJsonMapping has no additional fields

Reply:
Okay, then maybe your previous implementation was okay. 
Could you please get Sebastian's opinion also?

Reply:
Ok, I discussed this issue with Sebastian. He suggested to read: https://www.drdobbs.com/jvm/java-qa-how-do-i-correctly-implement-th/184405053 The article points out symmetry problems with equals in inherited classes. To avoid those I made my classes final. That reduces complexity. The abstract class does not lead to problems, as it, can't be instanciated directly.

For the hashcode problem Sebastian suggested to add the class name to the hash. This slves the problem that two two classes extending the abstract class have an equal hash code if they don't have own fields.

Reply:
Thanks for clarifying it! :+1: